src/GafferUI/ViewportGadget.cpp

```diff
@@ -1369,6 +1373,39 @@ void ViewportGadget::trackDragIdle()
 	requestRender();
 }
 
+void ViewportGadget::updateMotionState( const DragDropEvent &event, bool initialEvent )
+{
+	// Every time we transition from coarse to fine motion (or the reverse) we
```

Comment:
(Chatted off-line) The long-short is that there is currently a bunch of logic in `ViewportGadget` that tracks the mouse events of its children to allow auto-scrolling as you drag towards the end of a graph, etc. This made getting to our desired separation somewhat more arduous that it may have first seemed, and we deemed it a bit too much work to do as part of this.